packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ClientMethodTemplate.java

```diff
@@ -345,11 +316,10 @@ protected static void applyParameterTransformations(JavaBlock function, ClientMe
      *
      * @param function The client method code block.
      * @param clientMethod The client method.
-     * @param autoRestMethodRetrofitParameters Rest API method parameters.
      */
-    protected static void convertClientTypesToWireTypes(JavaBlock function, ClientMethod clientMethod,
-        List<ProxyMethodParameter> autoRestMethodRetrofitParameters) {
-        for (ProxyMethodParameter parameter : autoRestMethodRetrofitParameters) {
+    protected static void convertClientTypesToWireTypes(JavaBlock function, ClientMethod clientMethod) {
+        final List<ProxyMethodParameter> proxyMethodParameters = clientMethod.getProxyMethod().getParameters();
```

Comment:
Conditional checks in this method are simplified, reduced variable scopes and making it extensible so derived templates can reuse or override parts of the conversion.